src/bootstrap/lib.rs

```diff
@@ -1538,18 +1538,7 @@ impl Build {
         if self.config.dry_run() {
             return;
         }
```

Comment:
So this `dry_run` is only difference.

Reply:
But we may break things if we remove it? I'm not sure we want to do this clean up, seems like these functions are distinct if similar today.